crashtracker/src/telemetry.rs

```diff
@@ -0,0 +1,313 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2023-Present Datadog, Inc.
+
+use std::collections::HashMap;
+use std::fmt::Write;
+use std::time::{self, SystemTime};
+
+use super::{CrashInfo, CrashtrackerConfiguration, CrashtrackerMetadata};
+use anyhow::Ok;
+use ddtelemetry::{
+    build_host,
+    data::{self, Application, LogLevel},
+    worker::http_client::request_builder,
+};
+
+struct TelemetryMetadata {
+    application: ddtelemetry::data::Application,
+    host: ddtelemetry::data::Host,
+    runtime_id: String,
+}
+
+macro_rules! parse_tags {
+    (   $tag_iterator:expr,
+        $($tag_name:literal => $var:ident),* $(,)?)  => {
+        $(
+            let mut $var: Option<&str> = None;
+        )*
+        for tag in $tag_iterator {
+            let Some((name, value)) = tag.as_ref().split_once(':') else {
+                continue;
+            };
+            match name {
+                $($tag_name => {$var = Some(value);}, )*
+                _ => {},
+            }
+        }
+
+    };
+}
+
+#[derive(Debug, serde::Serialize)]
+/// This struct represents the part of the crash_info that we are sending in the
+/// log `message` field as a json
+struct TelemetryCrashInfoMessage<'a> {
```

Comment:
Should the `additional_stacktraces` go in here as well?

Reply:
I'd prefer adding it in a followup PR